pkg/sys/vfs/fs.go

```diff
@@ -0,0 +1,333 @@
+/*
+Copyright © 2022 - 2025 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vfs
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/twpayne/go-vfs/v4"
+)
+
+const (
+	DirPerm        = os.ModeDir | os.ModePerm
+	FilePerm       = 0666
+	NoWriteDirPerm = 0555 | os.ModeDir
+	TempDirPerm    = os.ModePerm | os.ModeSticky | os.ModeDir
+)
+
+type FS interface {
```

Comment:
Can we move it to where it is going to be used?

Reply:
This is a tricky one, we can move it sys together with all the non typed functions, but then we will have to also implement some functions. Let me give it a try.

Reply:
Oh, it is because the interface is actually the first argument to many functions and is technically used here.

Reply:
Yeah, this is a little hard. I guess let's keep it as is for now and we can see whether we can improve it later. Sounds like a plan?

Reply:
Already moved it to system package now.